packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1235,13 +1235,23 @@ mixin TextSelectionDelegate {
   ///   updating the text editing state.
   void userUpdateTextEditingValue(TextEditingValue value, SelectionChangedCause cause);
 
+  /// Shows the text selection toolbar.
+  void showToolbar();
+
   /// Hides the text selection toolbar.
   ///
   /// By default, hideHandles is true, and the toolbar is hidden along with its
   /// handles. If hideHandles is set to false, then the toolbar will be hidden
   /// but the handles will remain.
   void hideToolbar([bool hideHandles = true]);
 
+  /// Toggles the visibility of the toolbar.
+  ///
+  /// If the toolbar is hidden calling this method should show the toolbar.
+  ///
+  /// If the toolbar is visible calling this method should hide the toolbar.
```

Comment:
Nit: are the "should"s necessary here? Are there common scenarios where it doesn't show/hide the toolbar? If yes, should we call this out? If no, consider simplifying:

```suggestion
  /// If the toolbar is hidden calling this method shows the toolbar.
  ///
  /// If the toolbar is visible calling this method hides the toolbar.
```

I don't feel strongly about this, feel free to skip.

Reply:
Will simplify, thanks!